### PR TITLE
Remove refs from Browser component

### PR DESCRIPTION
**Pull request by azangru**
## Type
- refactoring

## Related JIRA Issue(s)
ENSWBSITES-174

## Description
The task description says "Check why useRef is used in the Browser.tsx and remove it if it’s not required".

`useRef` was used for 2 purposes in the Browser component:

1) to hold reference to the html element into which genome browser renders the canvas element, and
2) to get reference to the last value of active genome id coming from the props

Regarding the first usage, it is still relevant (some React components, such as zmenu, may need to be able to access the wrapper around canvas directly to calculate the dimensions of the canvas), but the ref itself can be moved from `Browser` to `BrowserImage`.
Second usage seems to be redundant at the moment.

**Note:** `BrowserImage` contained code that communicated with genome browser using the old mechanism (sending custom events via a DOM element) to tell genome browser stuff about cogs. I've changed the communication method to the new one (via browser-messaging-service) and moved this code to `BrowserCogList` component.

## Views affected
Genome browser screen.

### CODE REVIEW

File: src/ensembl/src/content/app/browser/BrowserCogList.tsx [hunk 1]
```diff
@@ -55,6 +59,37 @@ const BrowserCogList = (props: BrowserCogListProps) => {
...
   }, []);
 
+  useEffect(() => {
+    if (props.browserCogTrackList) {
+      const ons: string[] = [];
+      const offs: string[] = [];
+
+      /* what the frontend and backend call labels and names is flipped */
+      Object.keys(props.browserCogTrackList).forEach((name) => {
+        /* undefined means not seen means on for names */
```
Comment (imran-ebi) [inline src/ensembl/src/content/app/browser/BrowserCogList.tsx]: This comment is a bit cryptic. Can this be rephrased?

File: src/ensembl/src/content/app/browser/BrowserCogList.tsx [hunk 2]
```diff
@@ -55,6 +59,37 @@ const BrowserCogList = (props: BrowserCogListProps) => {
...
+        if (props.trackConfigNames[name]) {
+          ons.push(`${name}:label`);
+        } else {
+          offs.push(`${name}:label`);
+        }
+        /* undefined means not seen means off for labels */
```
Comment (imran-ebi) [inline src/ensembl/src/content/app/browser/BrowserCogList.tsx]: Same as above

File: src/ensembl/src/content/app/browser/BrowserCogList.tsx (hunk 1 above)
Comment (azangru) [reply src/ensembl/src/content/app/browser/BrowserCogList.tsx]: I completely agree, but that's not my comment; it came along with the code copied from BrowserImage. I will ask @ens-ds23 what he meant 😄

Comment (azangru) [reply src/ensembl/src/content/app/browser/BrowserCogList.tsx]: @imran-ebi I have talked to Dan and updated the comments. I hope they make a bit more sense now.

Comment (imran-ebi) [reply src/ensembl/src/content/app/browser/BrowserCogList.tsx]: Yes that's much better.

**Comment by imran-ebi** [comment]
Is there anything else that needs to be done in this PR?

**Pull request merged**